python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/resources.py

```diff
@@ -255,16 +265,58 @@ def load_check_specs(
         ]
 
     def cli(
-        self, args: Sequence[str], dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+        self,
+        args: Sequence[str],
+        dagster_dbt_translator: Optional[DagsterDbtTranslator] = None,
+        context: Optional[AssetExecutionContext] = None,
     ) -> DbtCloudCliInvocation:
         """Creates a dbt cli invocation with the dbt Cloud client."""
+        dagster_dbt_translator = validate_opt_translator(dagster_dbt_translator)
         dagster_dbt_translator = dagster_dbt_translator or DagsterDbtTranslator()
+
+        client = self.get_client()
         workspace_data = self.fetch_workspace_data()
+        job_id = workspace_data.job_id
+        manifest = workspace_data.manifest
+
+        assets_def: Optional[AssetsDefinition] = None
+        with suppress(DagsterInvalidPropertyError):
+            assets_def = context.assets_def if context else assets_def
```

Comment:
this is pretty weird... why is the None getting hidden behind the call to assets_def.
Is this how we translate it in dbt core as well? It feels pretty strange to me. What's even the context that this can be none? direct invocation I guess?

Reply:
Yes this reproduces the behavior in dbt core - it's to support direct invocation.

Reply:
weird... I'm pretty sure the assets def should be automatically resolved in direct invocation. But something we should solve in both places I guess